stable/sumologic-fluentd/templates/rbac.yaml

```diff
@@ -0,0 +1,24 @@
+{{ if .Values.rbac.enabled }}
+{{- $serviceName := include "fullname" . -}}
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/{{ required "A valid .Values.rbac.apiVersion entry required!" .Values.rbac.apiVersion }}
+metadata:
+  name: {{ $serviceName }}
+rules:
+- apiGroups: [""]
+  resources: ["namespaces", "pods"]
+  verbs: ["get", "list", "watch"]
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/{{ required "A valid .Values.rbac.apiVersion entry required!" .Values.rbac.apiVersion }}
+metadata:
+  name: {{ $serviceName }}
+subjects:
+- kind: ServiceAccount
+  name: {{ $serviceName }}
+  namespace: {{ .Release.Namespace }}
+roleRef:
+  kind: ClusterRole
+  name: {{ $serviceName }}
+  apiGroup: rbac.authorization.k8s.io
+{{ end }}
```

Comment:
Take a look at #1504 and use its RBAC pattern. Specifically:
  - One manifest/file per resource
  - Add common labels (app, chart, heritage, resource)
  - `name: {{ template "fullname" . }}`
  - `apiVersion: rbac.authorization.k8s.io/v1beta1` (since we currently only need to support K8s 1.6 & 1.7)
  - Create serviceAccount if `rbac.create=true`
  - Deployment should use created `serviceAccountName` if `rbac.create=true`. Otherwise use `rbac.serviceAccountName` (which must default to `default`)